### PR TITLE
added Views class (abstract and implementation)

### PR DESCRIPTION
**Pull request by peterBrxwn**

### CODE REVIEW

File: lib/views.dart [hunk 1]
```diff
@@ -0,0 +1,13 @@
+abstract class Views {
+  void stopViewById(String viewID, Map<String, Object> segmentation);
+
+  void stopLastView();
```
Comment (ArtursKadikis) [inline lib/views.dart:4]: for now not needed

File: lib/countly_config.dart [hunk 2]
```diff
@@ -278,4 +290,16 @@ class CountlyConfig {
     _remoteConfigValueCaching = true;
     return this;
   }
+
+  /// Used to enable automatic view tracking
+  CountlyConfig enableAutomaticViewTracking() {
```
Comment (ArtursKadikis) [inline lib/countly_config.dart:295]: not needed

File: lib/countly_config.dart [hunk 3]
```diff
@@ -278,4 +290,16 @@ class CountlyConfig {
...
+    _automaticViewTracking = true;
+    return this;
+  }
+
+  /// Used to enable global view segmentation
+  CountlyConfig enableGlobalViewSegmentation() {
```
Comment (ArtursKadikis) [inline lib/countly_config.dart:301]: wrong name and function

File: lib/countly_config.dart [hunk 4]
```diff
@@ -33,6 +33,10 @@ class CountlyConfig {
   bool _remoteConfigAutomaticTriggers = false;
   final List<RCDownloadCallback> _remoteConfigGlobalCallback = [];
   bool _remoteConfigValueCaching = false;
+  bool _automaticViewTracking = false;
```
Comment (ArtursKadikis) [inline lib/countly_config.dart:36]: not all fields are needed

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
segmentation values are optional for start/stop. not nullable

**Comment by peterBrxwn** [comment]
> segmentation values are optional for start/stop. not nullable

If the values are optional, it means the must be nullable (because if the are not passed, then their value is null)

File: lib/countly_config.dart [hunk 5]
```diff
@@ -99,6 +102,12 @@ class CountlyConfig {
 
   bool get remoteConfigValueCaching => _remoteConfigValueCaching;
 
+  bool get globalViewSegmentation => _globalViewSegmentation;
```
Comment (ArtursKadikis) [inline lib/countly_config.dart:105]: it's not a boolean

File: lib/countly_config.dart [hunk 6]
```diff
@@ -278,4 +287,10 @@ class CountlyConfig {
     _remoteConfigValueCaching = true;
     return this;
   }
+
+  /// Used to enable global view segmentation
+  CountlyConfig setGlobalViewSegmentation() {
```
Comment (ArtursKadikis) [inline lib/countly_config.dart:292]: it takes a map from string to object

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
Start adding the native side

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
"useMultipleViewFlow"
deprecate more old calls

"views_pauseViewWithID" -> "[Views] pauseViewWithID"

"startViewSegmentation" -> "startViewWithSegmentation"

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
"useMultipleViewFlow"
deprecate more old calls

"views_pauseViewWithID" -> "[Views] pauseViewWithID"

"startViewSegmentation" -> "startViewWithSegmentation"

**Pull request merged**